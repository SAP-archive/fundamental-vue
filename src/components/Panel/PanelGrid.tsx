import { Component, DefaultSlot, Prop, Base } from '@/core';

const colMapping = {
  2: '2-Column Grid',
  3: '3-Column Grid',
  4: '4-Column Grid',
  5: '5-Column Grid',
  6: '6-Column Grid',
};
type Col = keyof typeof colMapping;
const Cols = Object.keys(colMapping).map(value => Number(value)) as Col[];

interface Props {
  col?: number | null;
  nogap?: boolean;
}

@Component('PanelGrid')
@DefaultSlot('Panels displayed by the grid.')
export class PanelGrid extends Base<Props> {
  @Prop('number of columns', {acceptableValues: Cols, type: Number, default: null })
  public col!: Col | null;

  @Prop('whether there is a gap between the individual panels', { type: Boolean, default: false })
  public nogap!: boolean;

  public render() {
    return <div class={this.classes}>{this.$slots.default}</div>;
  }

  private get classes() {
    const col = this.col;
    const colClass = col == null ? {} : { [`fd-panel-grid--${col}col`]: true };
    return {
      'fd-panel-grid': true,
      'fd-panel-grid--nogap': this.nogap,
      ...colClass,
    };
  }
}
