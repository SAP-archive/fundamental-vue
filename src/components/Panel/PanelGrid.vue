<template>
  <div :class="classes">
    <slot />
  </div>
</template>

<script>
const PANEL_GRID_CLASS = "fd-panel-grid";

export default {
  name: "FdPanelGrid",
  props: {
    col: {
      type: Number,
      default: null,
      validator: value => value >= 2 && value <= 6
    },
    nogap: { type: Boolean, default: false }
  },
  computed: {
    classes() {
      const col = this.col;
      return [
        PANEL_GRID_CLASS,
        ...(col != null ? [`${PANEL_GRID_CLASS}--${col}col`] : []),
        ...(this.nogap ? [`${PANEL_GRID_CLASS}--nogap`] : [])
      ];
    }
  }
};
</script>
