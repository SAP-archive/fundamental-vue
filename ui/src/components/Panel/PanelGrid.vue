<template>
  <div :class="classes">
    <slot />
  </div>
</template>

<script lang="ts">
const PANEL_GRID_CLASS = "fd-panel-grid";

import Vue from "vue";
export default Vue.extend({
  name: "FdPanelGrid",
  props: {
    col: {
      type: Number,
      default: null,
      validator: (value: number) => value >= 2 && value <= 6
    },
    nogap: { type: Boolean, default: false }
  },
  computed: {
    classes(): string[] {
      const col = this.col;
      return [
        PANEL_GRID_CLASS,
        ...(col != null ? [`${PANEL_GRID_CLASS}--${col}col`] : []),
        ...(this.nogap ? [`${PANEL_GRID_CLASS}--nogap`] : [])
      ];
    }
  }
});
</script>
