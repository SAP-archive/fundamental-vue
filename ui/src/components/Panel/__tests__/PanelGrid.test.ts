import { shallowMount } from "@vue/test-utils";
import PanelGrid from "../PanelGrid.vue";

describe("PanelGrid", () => {
  const col = 4;
  const nogap = true;
  const defaultSlot = "Default slot";

  const nogapPanelGrid = shallowMount(PanelGrid, {
    propsData: {
      col,
      nogap
    },
    slots: {
      default: defaultSlot
    }
  });
  const panelGrid = shallowMount(PanelGrid, {
    propsData: {
      col: 2
    }
  });
  it("renders correctly", () => {
    expect(nogapPanelGrid.element).toMatchSnapshot();
  });
  it("renders default slot", () => {
    expect(nogapPanelGrid.find(".fd-panel-grid").text()).toBe(defaultSlot);
  });
  it("renders correctly nogap:false ", () => {
    expect(panelGrid.element).toMatchSnapshot();
  });
});
