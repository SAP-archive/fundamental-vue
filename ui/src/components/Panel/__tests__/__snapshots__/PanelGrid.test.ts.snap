// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelGrid renders correctly 1`] = `
<div
  class="fd-panel-grid fd-panel-grid--4col fd-panel-grid--nogap"
>
  Default slot
</div>
`;

exports[`PanelGrid renders correctly nogap:false  1`] = `
<div
  class="fd-panel-grid fd-panel-grid--2col"
/>
`;
